qiskit/aqua/algorithms/phase_estimators/phase_estimator.py

```diff
@@ -0,0 +1,91 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Phase Estimator interface"""
+
+from typing import Union, Optional
+from abc import ABC, abstractmethod
+import numpy
+import qiskit.circuit as circuit
+from qiskit.circuit import QuantumCircuit
+from qiskit.aqua.algorithms import AlgorithmResult
+
+# pylint: disable=attribute-defined-outside-init
+
+
+class PhaseEstimator(ABC):
+    """The Phase Estimator interface
+
+    Algorithms that can compute a phase for a unitary operator and
+    initial state may implement this interface to allow different
+    algorithms to be used interchangeably.
+    """
+
+    @abstractmethod
+    def estimate(self,
```

Comment:
If a third-party was to implement a phase estimator for Aqua this represents the set of attributes that would be required? Since there seems to be some sort of implementation here is this truly generic or just a base class for the current implementation. I guess I am less keen on seeing these (an interface and a convenience base class) combined if it ends up as a constraint over any future other implementations. I think I might have argued here that an estimate should just take whatever we are doing the phase estimation for, other properties, like num_eval_qubits should be set via the algo setter if they are not generic enough. Compare to interface of min eigen solver for instance - where it just takes the operator on which the eigenvalue is computed - other properties are part of main algo and are set that way.
Maybe the params are ok, but I am left with the impression this a more a base class than an interface as such - especially when it seems to share variables with the implementations such that the pylint disable above seems to be needed.